### PR TITLE
resolve ambiguity in tarfs

### PR DESCRIPTION
**Pull request by fffw**
Fixes https://github.com/getlantern/lantern/issues/2552

### CODE REVIEW

**Comment by myleshorton** [comment]
Nice! That fixes it for me, although I have to admit I don't understand the change. Can you explain @fffw?

File: src/github.com/getlantern/tarfs/tarfs.go [hunk 1]
```diff
@@ -252,7 +252,7 @@ func (f *AssetFile) Readdir(count int) ([]os.FileInfo, error) {
 }
 
 func (f *AssetFile) Stat() (os.FileInfo, error) {
-	return f, nil
+	return &f.FakeFile, nil
```
Comment (fffw) [inline src/github.com/getlantern/tarfs/tarfs.go]: `AssetFile` embeds 3 members. Seems another member also has `Size()` method, which causes ambiguity, although I don't know exactly which one. Returning `&f.FakeFile` explicitly to eliminate the ambiguity, as `FakeFile` is existed to provided an `os.FileInfo` interface.

Comment (myleshorton) [inline src/github.com/getlantern/tarfs/tarfs.go]: Ah I think I see -- conflict between AssetFile and AssetDirectory correct?

File: src/github.com/getlantern/tarfs/tarfs.go [hunk 2]
```diff
@@ -274,7 +274,3 @@ func newAssetDirectory(name string) *AssetDirectory {
 func (f *AssetDirectory) Readdir(count int) ([]os.FileInfo, error) {
 	return emptyFileInfos, nil
 }
-
```
Comment (fffw) [inline src/github.com/getlantern/tarfs/tarfs.go]: Embed `AssetFile` already have `Stat()` method. See https://golang.org/doc/effective_go.html#embedding

**Comment by myleshorton** [comment]
LGTM.

File: src/github.com/getlantern/tarfs/tarfs.go (hunk 1 above)
Comment (fffw) [inline src/github.com/getlantern/tarfs/tarfs.go]: I don't think so, though actually I don't know what causes the ambiguity. Removing `AssetDirectory` altogether doesn't stop this error.

**Pull request merged**